src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/Msmq/MsmqCommon.cs

```diff
@@ -0,0 +1,54 @@
+using System;
+using System.Text;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Tagging;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Msmq
+{
+    internal class MsmqCommon
+    {
+        internal static readonly IntegrationInfo IntegrationId = IntegrationRegistry.GetIntegrationInfo(MsmqConstants.IntegrationName);
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(MsmqCommon));
+
+        internal static Scope CreateScope<TMessageQueue>(Tracer tracer, string command, string spanKind, TMessageQueue messageQueue, bool? messagePartofTransaction = null)
+            where TMessageQueue : IMessageQueue
+        {
+            if (!tracer.Settings.IsIntegrationEnabled(IntegrationId))
+            {
+                // integration disabled, don't create a scope, skip this trace
+                return null;
+            }
+
+            Scope scope = null;
+
+            try
+            {
+                var tags = new MsmqTags(spanKind)
+                {
+                    Command = command,
+                    IsTransactionalQueue = messageQueue.Transactional.ToString(),
+                    UniqueQueueName = messageQueue.FormatName,
+                };
+                if (messagePartofTransaction.HasValue)
+                {
+                    tags.MessageWithTransaction = messagePartofTransaction.ToString();
+                }
+
+                var serviceName = tracer.Settings.GetServiceName(tracer, MsmqConstants.ServiceName);
+
+                scope = tracer.StartActiveWithTags(MsmqConstants.OperationName, serviceName: serviceName, tags: tags);
+
+                var span = scope.Span;
+                span.Type = SpanTypes.Queue;
+                span.ResourceName = command;
```

Comment:
Copying my comments from [here](https://github.com/DataDog/dd-trace-dotnet/pull/1457/files#r631160043):
> What will the resource name look like? I tried looking in the tests and we're not asserting that there (we probably should).
>
>If users are sending messages to different queues, we want those grouped by queue name in the UI. Should we append the Path to the current resource name?. Something like `{command} {path}`? Similar to http requests which use `{verb} {path}` (e.g. `GET /users/edit/{id}`).

Reply:
oh I see now, yes 👍

Reply:
I saw you check the resource name in the tests, thanks! What do you think of the second part of the comment (adding the path to the resource name)?

Reply:
yes so the we would have a resource name like that then:

resource_name:"msmq.purge DIRECT=OS:desktop-duevvft\\Private$\\private-nontransactional-queue"
resource_name:"msmq.send DIRECT=OS:desktop-duevvft\\Private$\\private-transactional-queue"

I could bring back the Path property (and not the whole name with host) to get something shorter without the host name

resource_name:"msmq.purge \\Private$\\private-nontransactional-queue"
resource_name:"msmq.send \\Private$\\private-transactional-queue"

![image](https://user-images.githubusercontent.com/8353486/118018202-c8f5a300-b357-11eb-8889-cca5a056c4b7.png)

Reply:
I'm not familiar with MSMQ and what users would want to see in the resource name, but I'm leaning towards Path only instead of the whole name.

Reply:
In the end used command - path 
without name of the machine